server/src/main/java/org/candlepin/async/JobMessageReceiver.java

```diff
@@ -61,6 +61,7 @@ protected String getQueueAddress() {
     @Override
     public void onMessage(ClientMessage message) {
```

Comment:
This method could use some special attention during review. I'm not entirely confident this is the right way to handle everything.

Reply:
I think that the first thing we need to do is to define what behavior we want in particular scenarios.

In the happy day scenario, if job execution fails, then the manager handles everything internally, and the receiver is done with the message.

It gets tricky when system components start going awry. What happens when Artemis is down/restarted while jobs are in flight? What happens if we fail to commit an updated status to the DB (DB down, or another exception)?

If Artemis goes down before the artemis session is committed, how do we want to handle that case?

When Artemis is down, we are not able to send a retry message. Perhaps we could try to re-send the message N times before deeming we can not retry.

When there's an issue updating the JobStatus, things are even worse given where we could be in the job execution path. Do we want Artemis to re-send the message and re-execute the entire job? Will the job constraints block re-execution?

We should also consider whether we want Artemis to do any message re-sends at all, and auto acknowledge the message as soon as it hits the receiver. This leaves the door open for job status entries being in a stale/bad state though.

Overall, I think that this warrants a discussion because of all the edge cases.